### PR TITLE
[5.1] Use Dialog for Field Media and Editor Media button

### PR DESCRIPTION
**Pull request by Fedik**
Pull Request for Issue # .

### Summary of Changes

Update Field Media, and Editor media button to use JoomlaDialog.

### Testing Instructions
Run `npm install`
Edit article and use Media button to insert image to article,
Also try edit "Intro Image" of the article

### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed
- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

References:
- https://github.com/joomla/joomla-cms/pull/40150
- https://github.com/joomla/joomla-cms/pull/40202

### CODE REVIEW

**Comment by Fedik** [comment]
Will rebase it later

File: layouts/joomla/form/field/media.php [hunk 1]
```diff
@@ -84,41 +84,38 @@
         $src = '';
     }
 
-    $width = $previewWidth;
+    $width  = $previewWidth;
     $height = $previewHeight;
-    $style = '';
+    $style  = '';
     $style .= ($width > 0) ? 'max-width:' . $width . 'px;' : '';
```
Comment (Quy) [inline layouts/joomla/form/field/media.php]:
```suggestion
    $style  = ($width > 0) ? 'max-width:' . $width . 'px;' : '';
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on bf74e2005341feccb630309fd4d0a8c34561ba6e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42288">issues.joomla.org/tracker/joomla-cms/42288</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on bf74e2005341feccb630309fd4d0a8c34561ba6e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42288">issues.joomla.org/tracker/joomla-cms/42288</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42288">issues.joomla.org/tracker/joomla-cms/42288</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @Fedik and also to the testers @Quy @viocassel ! 🎉

**Pull request merged**